go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -96,7 +96,15 @@ func getEnvGoSemVer() string {
 	if !strings.HasPrefix(goVersion, "go") {
 		log.Fatalf("Expected 'go version' output of the form 'go1.2.3'; got '%s'", goVersion)
 	}
-	return "v" + goVersion[2:]
+	// Go versions don't follow the SemVer format, but the only exception we normally care about
+	// is release candidates; so this is a horrible hack to convert e.g. `go1.22rc1` into `go1.22-rc1`
+	// which is compatible with the SemVer specification
+	rcIndex := strings.Index(goVersion, "rc")
+	if rcIndex != -1 {
+		return semver.Canonical("v"+goVersion[2:rcIndex]) + "-" + goVersion[rcIndex:]
```

Comment:
I find it confusing that we call `semver.Canonical` here and also on line 798 (but not at other call-sites). Is that intentional?

Reply:
I think L798 is leftover from the first attempt at getting this to work and is no longer necessary, so can be removed.

Reply:
That makes sense. Would there be any benefit in also using `semver.Canonical` on the other branch, so that we know we're always returning a canonical version?

Reply:
Done